server/WFCORE-6728-reload-to-stability-level.adoc

```diff
@@ -0,0 +1,155 @@
+# Add any category for this proposal
+# if missing, add it to _data/widfly-categories and use its id
+---
+= Ability to reload a server to a different stability level
+:author:            Kabir Khan
+:email:             kkhan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WildFly's (and WildFly Core's) testsuite runs at a particular stability level by default. At the time of writing this
+is set to the 'community' level.
+
+Since features may be introduced to WildFly at a lower stability level than this (e.g. 'experimental' and 'preview'), those features will not be available at the stability level the tests are currently run. Thus, we need a way to run tests at a chosen stability level.
+
+This proposal attempts to solve that by providing a set of ServerSetupTasks that will reload the server to the desired stability level before running the test. To support this a new management operation based on the existing `reload` operation will be introduced.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE[WFCORE-6728]
+
+=== Related Issues
+
+//* https://issues.redhat.com/browse/WFLY[WFLY-XXXX]
+
+=== Stability Level
+// Choose the planned stability level for the proposed functionality
+* [ ] Experimental
+
+* [ ] Preview
+
+* [x] Community
+
+* [ ] default
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly Core. The main changes will be in the server, testsuite/shared and testsuite/testsuite-runner modules.
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The WildFly Core test runner is enhanced to handle JUnit `Assume.assumeXXX()` calls from ServerSetupTasks the same way as in normal test code. Prior to this RFE the resulting `AssumptionViolatedException` would result in a stacktrace and test error, rather than the test being marked as skipped.
+* There will be a new operation based on the current reload operation introduced for standalone servers.
+- The operation will be registered at stability level 'community' at present
+- The operation is based on the reload operation, and adds a parameter called `stability`. This parameter takes the stability level we want to reload the server to
+- The operation also supports all the parameters supported by the standard reload operation
+- The operation has its own RBAC setting to narrow down who can call it
```

Comment:
Please specify the constraint name and the OOTB setting for this.

The constraint name is kind of an impl detail, but since you know it, it's helpful to reviewers to write it down.